src/LocalStorageStore.ts

```diff
@@ -0,0 +1,54 @@
+import Store from "./Store";
+import MemoryStore from "./MemoryStore";
+
+/** Class that stores key/value pairs in localStorage if possible
+    but falls back to an in-memory store. */
+export default class LocalStorageStore implements Store {
+    private enabled: boolean;
+    private memoryStore: MemoryStore | null;
+    private manifestUrl: string;
+    
+    public constructor() {}
+
+    public async start(manifestUrl: string): Promise<void> {
+        this.manifestUrl = manifestUrl;
+        try {
+            // In some browsers (eg iOS Safari in private mode), 
+            // localStorage exists but throws an exception when
+            // you try to write to it.
+            window.localStorage.setItem("test", "test");
```

Comment:
Just to be an extra 0.001% better, maybe we can store a unique string instead (e.g. a UUID). This would prevent overwriting something that someone named `test`, of course.